scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -0,0 +1,348 @@
+        content = setEntity(content, selection, afterKey, 'DELETE_SUGGESTION');
+    } else {
+        content = addNewSuggestionEntity(content, selection, 'DELETE_SUGGESTION', data);
+    }
+
+    newState = EditorState.push(newState, content, 'insert-suggestions');
+    newState = RichUtils.toggleInlineStyle(newState, 'DELETE_SUGGESTION');
+    return changeEditorSelection(newState, 0, -1, true);
+};
+
+/**
+ * @ngdoc method
+ * @name applyStyleForSuggestion
+ * @param {Object} editorState
+ * @param {Objest} inlineStyle
+ * @param {String} type
+ * @return {Object} returns new state
+ * @description Apply the style for current selection.
+ */
+const applyStyleForSuggestion = (editorState, inlineStyle, type) => {
+    let newState = editorState;
+
+    inlineStyle.forEach((style) => {
+        if (style != 'ADD_SUGGESTION' && style != 'DELETE_SUGGESTION') {
+            newState = RichUtils.toggleInlineStyle(newState, style);
+        }
+    });
+
+    return RichUtils.toggleInlineStyle(newState, type);
+};
+
+/**
+ * @ngdoc method
+ * @name resetDefaultSugestionStyle
+ * @param {Object} editorState
+ * @return {Object} returns new state
+ * @description Filter out suggestion related styles from current styles for next character.
+ */
+const resetDefaultSugestionStyle = (editorState) => {
+    const types = ['DELETE_SUGGESTION', 'ADD_SUGGESTION'];
+    let inlineStyle = editorState.getCurrentInlineStyle();
+
+    inlineStyle = inlineStyle.filter((style) => types.indexOf(style) === -1);
+    return EditorState.setInlineStyleOverride(editorState, inlineStyle);
+};
+
+/**
+ * @ngdoc method
+ * @name resetSuggestion
+ * @param {Object} editorState
+ * @param {String} type
+ * @return {Object} returns new state
+ * @description For type suggestion reset both style and entity for
+ * current character position.
+ */
+const resetSuggestion = (editorState, type) => {
+    let newState = editorState;
+    let selection;
+    let content;
+
+    newState = changeEditorSelection(newState, 0, 1, false);
+    newState = RichUtils.toggleInlineStyle(newState, type);
+
+    content = newState.getCurrentContent();
+    selection = newState.getSelection();
+
+    content = deleteEntity(content, selection, type);
+    newState = EditorState.push(newState, content, 'reset-suggestions');
+    newState = changeEditorSelection(newState, 1, 0, false);
+
+    return newState;
+};
+
+/**
+ * @ngdoc method
+ * @name deleteCharacter
+ * @param {Object} editorState
+ * @return {Object} returns new state
+ * @description Delete the current character.
+ */
+const deleteCharacter = (editorState) => {
+    let content;
+    let selection;
+    let newState;
+
+    newState = changeEditorSelection(editorState, -1, 0, false);
+    content = newState.getCurrentContent();
+    selection = newState.getSelection();
+
+    content = Modifier.removeRange(content, selection, 'forward');
+    newState = EditorState.push(newState, content, 'delete-suggestions');
```

Comment:
the change types are given by draft - https://github.com/facebook/draft-js/blob/master/src/model/immutable/EditorChangeType.js